rust/chains/abacus-ethereum/src/trait_builder.rs

```diff
@@ -27,6 +30,27 @@ pub trait MakeableWithProvider {
         metrics: Option<(ProviderMetrics, PrometheusMiddlewareConf)>,
     ) -> eyre::Result<Self::Output> {
         Ok(match conn {
+            Connection::HttpQuorum { urls, weights } => {
+                let mut builder = QuorumProvider::builder().quorum(Quorum::Majority);
+                for (i, url) in urls.into_iter().enumerate() {
+                    let http_provider = url.parse::<Http>()?;
+                    let weight = weights
+                        .as_ref()
+                        .map(|weights| {
+                            weights
+                                .get(i)
+                                .ok_or(eyre!("Expected one weight for every url provided"))
+                        })
+                        .transpose()?
+                        .cloned()
+                        .unwrap_or(1);
+                    let weighted_provider = WeightedProvider::with_weight(http_provider, weight);
+                    builder = builder.add_provider(weighted_provider);
+                }
+                // let retrying = RetryingProvider::new(builder.build(), 6, 50);
```

Comment:
nit: rm commented code?

Also wanna make sure I'm reading this right: we're *not* wrapping the QuorumProvider with a RetryingProvider atm right? Should we be wrapping it?

Reply:
We are now. And yes this was a mistake.